failure in building man-pages as part of the docs (HTML version)

steps to reproduce:
```bash
./configure --enable-manpages-in-doc
make
```
failure output:
```
dir=doc/man-html/ && \
/bin/mkdir -p $dir && \
PYTHONPATH=. ENABLE_MANPAGES=1 COPY_DOC=1 \
        HTML_THEME=classic \
./autotools/run-in-tempdir \
    /root/ganeti/autotools/sphinx-wrapper /usr/bin/sphinx-build -q -W -b html \
        -d . \
        -D version="3.0" \
        -D release="3.0.0~alpha1" \
        -D graphviz_dot="/usr/bin/dot" \
doc /root/ganeti/$dir && \
rm -f $dir/.buildinfo $dir/objects.inv

Warning, treated as error:
/tmp/gntbuild.DMIzwKix/doc/man-ganeti.rst:680:Inline strong start-string without end-string.
make: *** [Makefile:4228: doc/man-html/index.html] Error 2
```

when #1486 is applied, the building of the html manpages steps further, but now stops with reference target not found:
```shell
dir=doc/man-html/ && \
/bin/mkdir -p $dir && \
PYTHONPATH=. ENABLE_MANPAGES=1 COPY_DOC=1 \
        HTML_THEME=classic \
./autotools/run-in-tempdir \
    /tmp/d4ba7e040ed24b1fbed2a6224b5c4fb2c93f7603/autotools/sphinx-wrapper /usr/bin/sphinx-build -q -W -b html \
	-d . \
	-D version="3.0" \
	-D release="3.0.0~beta1" \
	-D graphviz_dot="/usr/bin/dot" \
doc /tmp/d4ba7e040ed24b1fbed2a6224b5c4fb2c93f7603/$dir && \
rm -f $dir/.buildinfo $dir/objects.inv
[GHC]: src/hluxid.o <- cabal_macros.h src/hluxid.hs src/Ganeti/Runtime.hi src/Ganeti/Runtime.o src/Ganeti/Daemon.hi src/Ganeti/Daemon.o src/Ganeti/Query/Server.hi src/Ganeti/Query/Server.o
compilation IS NOT required
[GHC-link]: src/hluxid
/usr/bin/ghc -O -Wall -isrc -fwarn-monomorphism-restriction -fwarn-tabs -optP-include -optPcabal_macros.h -hide-all-packages `cat ganeti.depsflags` -fwarn-incomplete-uni-patterns -fhpc -DTEST -dynamic-too \
	  --make src/hluxid.hs
Linking src/hluxid ...
cp -pf src/hluxid src/ganeti-luxid
PYTHONPATH=. ./autotools/run-in-tempdir \
  /tmp/d4ba7e040ed24b1fbed2a6224b5c4fb2c93f7603/./autotools/build-bash-completion --compact > doc/examples/bash_completion
PYTHONPATH=. ./autotools/run-in-tempdir \
  /tmp/d4ba7e040ed24b1fbed2a6224b5c4fb2c93f7603/./autotools/build-bash-completion  > doc/examples/bash_completion-debug

Warning, treated as error:
/tmp/gntbuild.G9ThAeNl/doc/admin.rst:161:'doc' reference target not found: man-ganeti-extstorage-interface
make: *** [Makefile:4230: doc/man-html/index.html] Error 2
make: *** Deleting file 'doc/man-html/index.html'
```

--- a/lib/build/sphinx_ext.py
+++ b/lib/build/sphinx_ext.py
@@ -360,6 +360,7 @@ def _ManPageNodeClass(*args, **kwargs):
 
   # Force custom title
   kwargs["refexplicit"] = True
+  kwargs["refdomain"] = None
 
   return sphinx.addnodes.pending_xref(*args, **kwargs)
 

